### PR TITLE
Task/refactor http headers

### LINKED ISSUE #141: Refactor common code in CgiHandler and HttpRequest
There is some code duplication between these classes.

We could extract the parsing of HTTP headers to a new class/component.

### PR DESCRIPTION
**Pull request by gbudau**
closes #141

### CODE REVIEW

File: srcs/app/http/HttpHeaders.cpp [hunk 1]
```diff
@@ -0,0 +1,108 @@
+#include <HttpHeaders.hpp>
+#include <cctype>
```
Comment (Damrod) [inline srcs/app/http/HttpHeaders.cpp:2]: Do you agree that it is a good idea to place the additional includes for a source file in the header of its name?
I believed it makes code cleaner, but I've come across people with different opinions.

**Review by Damrod — APPROVED** [review]
Looks great, thank you!

File: srcs/app/http/HttpHeaders.cpp (hunk 1 above)
Comment (gbudau) [reply srcs/app/http/HttpHeaders.cpp:2]: I prefer to include only the necessary headers in the header files and the rest in the source files. But I see that we place all additional includes in the header files, and I will adhere to the same practice for consistency.

**Review by vdedios — APPROVED** [review]
:+1: :+1: :+1:

**Comment by gbudau** [comment]
Thank you! :raised_hands:

**Pull request merged**